packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -561,7 +567,15 @@ export const Table = <
         selectRowOnClick &&
         !event.isDefaultPrevented()
       ) {
-        if (!row.isSelected && (selectionMode === 'single' || !event.ctrlKey)) {
+        if (event.shiftKey) {
```

Comment:
To prevent text selection, there are two approaches. Both of these would either go in the row onclick or keydown and should only be done when `selectionMode` is set to `'multi'`.
- use `getSelection()` to temporarily store the old selection, call `removeAllRanges` on the selection to clear selection, then set the selection to old selection (effectively preserving user's selection state)
- or, apply `user-select: none` on the table when shift selection starts and remove it when it's done

these are all standard platform APIs so you can find more info on MDN

Reply:
Hmm this is causing some flickering when the text selection gets cleared. Not a very nice user experience.

Reply:
**[DONE]** Ohh, right. @bentleyvk implemented a much easier way ([code lines](https://github.com/iTwin/iTwinUI-react/compare/vilius/shift-select-demo#diff-9edd44b431060c27e0260d286ee188bdf901225b1ee84ff6540f9d796dd0f999R778-R786)) using the `user-select: none` way you suggested. I added that here. It sets `user-select: none` when shift is pressed and removes `user-select` when shift is released. It works without flickering.

Also one thing I noticed: when we scroll out of the selection in a table with virtualization, then the selection is lost (end of video). I'm sure this might be okay since that is how virtualization works, but just noting what I noticed.

[vokoscreenNG-2022-09-02_14-46-33.webm](https://user-images.githubusercontent.com/45748283/188218865-42110b5b-f5f9-46c3-bd68-5b7bf024e575.webm)